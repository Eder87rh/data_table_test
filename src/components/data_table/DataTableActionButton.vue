<template>
    <div>
        <button class="button" 
            @click="action"
        >
            {{ this.title.toUpperCase() }}
        </button>
    </div>
</template>

<script>
export default {
    props:{
        title:{
            required:true,
            type:String
        },
        action:{
            required:true,
            type:Function,
        }
    }
}
</script>

<style lang="scss" scoped>

</style>
