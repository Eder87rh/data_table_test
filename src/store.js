import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    data:[
      {
        id: '3471DA17-401F-9633-BF81-4CADA6FD5C79',
        name: 'Kyra Lester',
        description: 'Curabitur dictum. Phasellus in',
        date: '2017-07-23T04:24:49-07:00',
        amount: '345.54'
      },
      {
          id: '3471DA17-401F-9633-BF81-4CADA6FD5C79',
          name: 'Kyra Lester',
          description: 'Curabitur dictum. Phasellus in',
          date: '2017-07-23T04:24:49-07:00',
          amount: '345.54'
      }
    ]
  },
  mutations: {

  },
  actions: {
    
  }
})
