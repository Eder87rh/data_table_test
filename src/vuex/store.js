import Vue from 'vue'
import Vuex from 'vuex'
import modPayments from './modules/payments'
import moodFilters from './modules/filters'

Vue.use(Vuex)

export default new Vuex.Store({
  modules:{
    modPayments,
    moodFilters,
  },
  state: {
    
  },
  mutations: {

  },
  actions: {

  }
})
