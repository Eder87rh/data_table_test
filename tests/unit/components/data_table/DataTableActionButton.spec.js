import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import DataTableActionButton from '@/components/data_table/DataTableActionButton.vue'
import store from '@/vuex/store.js'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('DataTableActionButton.vue', () => {
    it('match snapshot',() => {
        const wraper = shallowMount(DataTableActionButton, {
            mocks: {
                $store: store
            },
            propsData:{
                title:'Save to Firestore',
                action:() =>{}
            },
            localVue
        })
        expect(wraper.html()).toMatchSnapshot()
    })
})