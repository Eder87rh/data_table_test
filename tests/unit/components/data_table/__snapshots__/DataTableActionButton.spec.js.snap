// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableActionButton.vue match snapshot 1`] = `
<div><button class="button">
    SAVE TO FIRESTORE
  </button></div>
`;
